pwndbg/commands/__init__.py

```diff
@@ -276,21 +277,26 @@ def _OnlyWithResolvedHeapSyms(*a, **kw):
         e = lambda s: print(message.error(s))
         w = lambda s: print(message.warn(s))
         if pwndbg.heap.current.can_be_resolved():
+            # Note: We will still raise the error for developers when exception-* is set to "on"
             try:
                 return function(*a, **kw)
             except SymbolUnresolvableError as err:
                 e(f"{function.__name__}: Fail to resolve the symbol: `{err.symbol}`")
                 w(
                     f"You can try to determine the libc symbols addresses manually and set them appropriately. For this, see the `heap_config` command output and set the config about `{err.symbol}`."
                 )
+                if pwndbg.config.exception_verbose.value or pwndbg.config.exception_debugger.value:
+                    raise err
             except Exception as err:
-                e(f"{function.__name__}: Unknown error: `{err}` when running this symbols.")
+                e(f"{function.__name__}: An unknown error occurred when running this command.")
                 if pwndbg.config.resolve_heap_via_heuristic:
                     w(
                         "Maybe you can try to determine the libc symbols addresses manually, set them appropriately and re-run this command. For this, see the `heap_config` command output and set the `main_arena`, `mp_`, `global_max_fast`, `tcache` and `thread_arena` addresses."
                     )
                 else:
                     w("You can try `set resolve-heap-via-heuristic on` and re-run this command.\n")
+                if pwndbg.config.exception_verbose.value or pwndbg.config.exception_debugger.value:
```

Comment:
```suggestion
                if pwndbg.config.exception_verbose or pwndbg.config.exception_debugger:
```

Reply:
The `pwndbg.config.Parameter` class has a `__bool__` magic method/operator defined:
```py
    def __bool__(self):
        return bool(self.value)
```

so we can omit `.value` here :)